### ISSUE TITLE
Feature request: strict comparison inspector

### ISSUE THREAD

**Issue by elomonaco** [hammer-tools/php-hammer#8]
it'd be nice to have an inspection that'd inform the user that they should use the === instead of ==.

**Comment by rentalhost** [comment]
Thanks for the suggestion.

Until then I believed that PhpStorm performed this type of analysis on its own, but it seems not. Maybe I can apply this rule to cases where it's 100% easy to pay attention to the types of data being compared, for example:

```php
function test(int $x): bool {
     return $x == 123; // int == int, so int === int
}
```

**Comment by rentalhost** [comment]
**Implemented in version 0.22.0:**

Hello @elomonaco, the new **version 0.22.0** has just been released and is available for direct download from the **[Releases](https://github.com/hammer-tools/php-hammer/releases/tag/0.22.0)** page. This update will be available on the **[Jetbrains Market](https://plugins.jetbrains.com/plugin/19515-php-hammer)** in approximately three days if you prefer to wait.

**This version comes with some new features, now totaling 51 inspections:**

- Support for the **new PhpStorm 2023.2 EAP** (_minimum version from now on_);
- **Your suggestions:** inspection to suggest strict comparison when possible;
- **Your suggestions:** Inspection to suggest the use of non-deprecated functions (e.g., `mt_rand()` to `random_int()`);
- Additional feature: add missing _semicolon_ when saving a PHP file;
- Other minor improvements.

Thank you for taking the time to suggest these improvements and for patiently waiting for their implementation.
